File: src/testing/insertArticles2DB.js
const { MongoClient } = require("mongodb");

const uri = "mongodb://127.0.0.1:27017/";
const dbName = "newsdata";
const collectionName = "articledata";

// Create a new MongoClient
const client = new MongoClient(uri);

// Function to insert articles into the database
const insertArticles = async (articles) => {
  try {
    // Connect to the MongoDB server
    await client.connect();

    // Select the database and collection
    const db = client.db(dbName);
    const collection = db.collection(collectionName);

    // Insert each article into the collection
    for (let article of articles) {
      // Check if an article with the same link exists
      const existingArticle = await collection.findOne({ link: article.link });

      if (!existingArticle) {
        // Insert the article into the collection
        await collection.insertOne(article);
        console.log(`Inserted article: ${article.title}`);
      } else {
        console.log(`Article already exists: ${article.title}`);
      }
    }
  } catch (error) {
    console.error("Error inserting articles:", error);
  } finally {
    // Close the connection
    await client.close();
  }
};

module.exports = insertArticles;

File: src/testing/insertArticles2db.py
from pymongo import MongoClient

uri = "mongodb://127.0.0.1:27017/"
dbName = "newsdata"
collectionName = "articledata"

# Create a new MongoClient
client = MongoClient(uri)

def insert_articles(articles):
    try:
        # Connect to the MongoDB server
        print("Connected to MongoDB")

        # Select the database and collection
        db = client[dbName]
        collection = db[collectionName]

        # Insert each article into the collection
        for article in articles:
            # Check if an article with the same link exists
            existing_article = collection.find_one({'link': article['link']})

            if not existing_article:
                # Insert the article into the collection
                collection.insert_one(article)
                print(f"Inserted article: {article['title']}")
            else:
                print(f"Article already exists: {article['title']}")

    except Exception as e:
        print("Error inserting articles:", e)

    finally:
        # Close the connection
        client.close()
        print("MongoDB connection closed")
